var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -266,6 +266,17 @@ def install(self, spec, prefix):
         else:
             options.append('--with-suitesparse=0')
 
+        # zlib: configuring using '--with-zlib-dir=...' has some issues with
```

Comment:
@jedbrown or @BarrySmith -- can you take a look at this? Without it the build was failing on my Mac.

Reply:
I would say Spack should never build PETSc with zlib. It can't be used directly by PETSc, but is a dependency of some PETSc optional dependencies such as HDF5. So in order for PETSc configure to support `--download-hdf5`, it also needs a way to find (or download and install) zlib. If HDF5 is built in advance, PETSc doesn't need to know whether it supports zlib.